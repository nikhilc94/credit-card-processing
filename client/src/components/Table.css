table {
  border-collapse: collapse;
  width: 100%;
}

thead {
  background-color: rgb(216, 216, 216);
}

tbody {
  text-align: center;
}

td,
th {
  border: 1px solid rgb(187, 187, 187);
  font-weight: 400;
  font-size: 15px;
  padding: 8px;
  width: 25%;
  word-break: break-all;
}

@media only screen and (max-width: 768px) {
  td,
  th {
    font-size: 11px;
  }
}
