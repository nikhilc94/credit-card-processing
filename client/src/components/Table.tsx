import React, { useEffect } from "react";

import "./Table.css";
import fetchCards from "../utils/fetchCards";
import numberWithCommas from "../utils/numberWithCommas";

interface TableProps {
  cardDetails: {
    _id: string;
    name: string;
    cardNumber: string;
    cardLimit: Number;
    cardBalance: Number;
  }[];
  setCardDetails: Function;
}

const Table: React.FC<TableProps> = ({ cardDetails, setCardDetails }) => {
  useEffect(() => {
    fetchCards(setCardDetails);
    // eslint-disable-next-line
  }, []);

  return (
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Card Number</th>
          <th>Balance</th>
          <th>Limit</th>
        </tr>
      </thead>
      <tbody>
        {cardDetails.map(
          ({ _id, name, cardNumber, cardLimit, cardBalance }) => (
            <tr key={_id}>
              <td>{name}</td>
              <td>{cardNumber}</td>
              <td>{numberWithCommas(cardBalance)}</td>
              <td>{numberWithCommas(cardLimit)}</td>
            </tr>
          )
        )}
      </tbody>
    </table>
  );
};

export default Table;
