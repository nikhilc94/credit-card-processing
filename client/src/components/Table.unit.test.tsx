import React from "react";
import { shallow } from "enzyme";
import { createSerializer } from "enzyme-to-json";

import Table from "./Table";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

const testData = [
  {
    _id: "djedhednwjksnwjwhw_ww",
    name: "Test user",
    cardNumber: "4567834619",
    cardLimit: 3500,
    cardBalance: 0
  },
  {
    _id: "mnfgeyyyyyssnwjwhw3d",
    name: "Test user second",
    cardNumber: "0964387189",
    cardLimit: 10000,
    cardBalance: 0
  }
];

describe("Table component", () => {
  it("renders successfully", () => {
    const wrapper = shallow(
      <Table cardDetails={testData} setCardDetails={() => null} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
