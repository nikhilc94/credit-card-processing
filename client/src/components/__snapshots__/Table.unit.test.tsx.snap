// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders successfully 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Card Number
      </th>
      <th>
        Balance
      </th>
      <th>
        Limit
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="djedhednwjksnwjwhw_ww"
    >
      <td>
        Test user
      </td>
      <td>
        4567834619
      </td>
      <td>
        £0
      </td>
      <td>
        £3,500
      </td>
    </tr>
    <tr
      key="mnfgeyyyyyssnwjwhw3d"
    >
      <td>
        Test user second
      </td>
      <td>
        0964387189
      </td>
      <td>
        £0
      </td>
      <td>
        £10,000
      </td>
    </tr>
  </tbody>
</table>
`;
